features/proxy_config.feature

```diff
@@ -0,0 +1,28 @@
+@uses.config.contract_token
+Feature: Attach command when proxy is configured
+
+    @series.lts
+    Scenario Outline: Attach command when proxy is configured
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I launch a `focal` `proxy` machine
+        And I run `apt install squid -y` `with sudo` on the `proxy` machine
+        And I add this text on `/etc/squid/squid.conf` on `proxy` above `http_access deny all`:
+            """
+            acl all src 0.0.0.0\/0\nhttp_access allow all
+            """
+        And I run `systemctl restart squid.service` `with sudo` on the `proxy` machine
```

Comment:
We might want all of this as its own `When I launch a squid proxy` command if we think we'll have additional Scenarios with proxies. Though I suppose we can probably add most of the proxy integration tests to the end of this scenario.

Reply:
I don't think this is a bad idea at all. At my idea here was to make this generic enough that we could use it for any other case that we need a separate machine for testing.

Even though right now we could create the `I launch a squid proxy` step, I think there are still other test scenarios we may want to exercise in the future, such as toying with firewall rules. Because of that, I am inclined for now to keep those steps as is

Reply:
Totally agree with the utility of launching separate machines in general. This works for me :+1: